api/tests/unit/domain/usecases/link-user-to-session-certification-candidate_test.js

```diff
@@ -135,178 +95,152 @@ describe('Unit | Domain | Use Cases | link-user-to-session-certification-candida
-          const certificationCandidateRepository = _buildFakeCertificationCandidateRepository()
-            .withFindBySessionIdAndPersonalInfo({
-              args: {
-                sessionId,
-                firstName,
-                lastName,
-                birthdate,
-              },
-              resolves: [certificationCandidate],
-            })
-            .withFindOneBySessionIdAndUserId({ args: { sessionId, userId }, resolves: 'existingLinkedCandidateToUser' });
+        context('when the linked user is the same as the user being linked', () => {
 
-          const certificationCenter = domainBuilder.buildCertificationCenter({ sessionId, type: 'SUP' });
-          const certificationCenterRepository = _buildFakeCertificationCenterRepository()
-            .withGetBySessionId({ args: sessionId, resolves: certificationCenter });
+          it('should not create a link and return the matching certification candidate', async () => {
+            // given
+            const certificationCandidate = domainBuilder.buildCertificationCandidate({ userId });
+            const certificationCandidateRepository = _buildFakeCertificationCandidateRepository()
+              .withFindBySessionIdAndPersonalInfo({
+                args: {
+                  sessionId,
+                  firstName,
+                  lastName,
+                  birthdate,
+                },
+                resolves: [certificationCandidate],
+              });
 
-          // when
-          const err = await catchErr(linkUserToSessionCertificationCandidate)({
-            sessionId,
-            userId,
-            firstName,
-            lastName,
-            birthdate,
-            certificationCandidateRepository,
-            certificationCenterRepository,
-          });
+            const certificationCenter = domainBuilder.buildCertificationCenter({ sessionId, type: 'SUP' });
+            const certificationCenterRepository = _buildFakeCertificationCenterRepository()
+              .withGetBySessionId({ args: sessionId, resolves: certificationCenter });
 
-          // then
-          expect(err).to.be.instanceOf(UserAlreadyLinkedToCandidateInSessionError);
+            // when
+            const result = await linkUserToSessionCertificationCandidate({
+              sessionId,
+              userId,
+              firstName,
+              lastName,
+              birthdate,
+              sessionRepository,
+              certificationCandidateRepository,
+              certificationCenterRepository,
+            });
+
+            // then
+            expect(result).to.deep.equal(
+              new UserAlreadyLinkedToCertificationCandidate(),
+            );
+          });
         });
-      });
 
-      context('when the user is not linked to any candidate in this session', () => {
+        context('when the linked user is the not the same as the user being linked', () => {
 
```

Comment:
'when the linked user is ~~the~~ not the same as the user being linked'

Reply:
> Je me pose juste la question de savoir si le mot "accessible" est un mot qu'on utilise aujourd'hui au niveau métier ?.

Est-ce que "joinable" serait pas plus approprié?

Reply:
Ok